src/ddmd/lexer.d

```diff
@@ -253,6 +270,12 @@ class Lexer
         t.lineComment = null;
         while (1)
         {
+            // Return error token when end of buffer is reached unexpectedly.
+            if (p >= end) {
```

Comment:
BTW, dmd's compile speed has slowed down substantially in the recent past, due to an accumulation of barnacles. It's now slower than dmc++.